File: EQUALISE.cpp
#include <bits/stdc++.h>
#define ll long long int
#define mod ll(1e9+7)
using namespace std;
void Luffy()
{
    int A,B,mn,mx,flag=0;
    cin>>A>>B;
    mn=min(A,B);
    mx=max(A,B);
    while(mx>=mn){
        if(mx==mn){
            flag=1;
            break;
        }else{
            mn=(mn<<1);
        }
    }
    (flag==1)? cout<<"YES"<<endl : cout<<"NO"<<endl;
    
}
int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    ll t; cin>>t;
    while(t--)
        Luffy();
}

File: EQUALISE.java
import java.util.*;
import java.lang.*;
import java.io.*;
class Codechef
{
	public static void main (String[] args) throws java.lang.Exception
	{
		Scanner in = new Scanner(System.in);
		int t;
		t= in.nextInt();
		while(t-- > 0){
		    int a=in.nextInt();
		    int b=in.nextInt();
		    int min=Math.min(a,b);
		    int max=Math.max(a,b);
		    int flag=0;
		    while(min<=max){
		        if(min==max){
		            flag=1;
		            break;
		        }
		        min*=2;
		    }
		    if(flag==0){
		        System.out.println("NO");
		    }else{
		        System.out.println("YES");
		    }
		}
	}
}
